### PR TITLE
ENH: Maxwell filter component regularization

### PR DESCRIPTION
**Pull request by Eric89GXL**
Requires `testing` and `sample` dataset updates. @agramfort feel free to try the example and see if it's good enough for you now :)

cc @wronk feel free to have a look and review. Note that I refactored a bit to have chunked data processing instead of processing everything at once. This should reduce memory requirements by a factor of ~2.

### CODE REVIEW

File: mne/preprocessing/maxwell.py [hunk 1]
```diff
@@ -502,6 +579,7 @@ def _sss_basis(origin, coils, int_order, ext_order, mag_scale=100.,
                          for coil in coils])] = mag_scale
 
     # Compute internal/external basis vectors (exclude degree 0; L/RHS Eq. 5)
+    mu_0 = 4e-7 * np.pi  # magnetic permeability
```
Comment (wronk) [inline mne/preprocessing/maxwell.py]: Might add a note here that Elekta may not be using the actual irrational value. Samu's code used a truncated number. Near zero percent chance it matters, but a comment is cheap :p

**Comment by wronk** [comment]
LGTM once tests pass. Nice addition :)

Do we have a ballpark for what the shielding factors should be in the tests? I thought Samu usually talks about it as a factor that's around couple hundred. Maybe it's just a difference in terminology?

**Comment by Eric89GXL** [comment]
Must be. The values are right at the passing limit, including the tests of
MF-processed files, and our values are close the theirs.

File: mne/preprocessing/maxwell.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/preprocessing/maxwell.py]: Well technically our value isn't irrational either :) But I added a comment at the top saying that other implementations could use single precision or truncated constants leading to small differences.

Comment (agramfort) [inline mne/preprocessing/maxwell.py]: can you share the evoked plots for:

- original raw data
- maxfilter elekta output
- mne-pyton maxwell_filter output

?

Comment (Eric89GXL) [inline mne/preprocessing/maxwell.py]: raw data:
raw_data

with `maxwell_filter(raw, cross_talk=ctc_fname, calibration=fine_cal_fname)`:
maxwell_filtered_data


with an equivalent call to `maxfilter` as `maxfilter -f sample_audvis_raw.fif -o sample_audvis_raw_sss.fif -ctc ct_sparse_mgh.fif -cal sss_cal_mgh.dat -regularize in -autobad off -iterate 0 -bad 2443 1032 2313` (iterate and regularize are default values for my setup, not sure about MGH):
maxfilter_equivalent_data

**Comment by agramfort** [comment]
looks very convincing !

you'll need to rebase before merge.

awesome work guys !

**Comment by Eric89GXL** [comment]
@agramfort done

**Comment by agramfort** [comment]
thx heaps ! :beers:

**Pull request merged**